lib/bcdice/game_system/AlchemiaStruggle.rb

```diff
@@ -0,0 +1,353 @@
+# frozen_string_literal: true
+
+require "bcdice"
+
+module BCDice
+  module GameSystem
+    class AlchemiaStruggle < Base
+      ID = "AlchemiaStruggle"
+      NAME = "アルケミア・ストラグル"
+      SORT_KEY = "あるけみあすとらくる"
+
+      HELP_MESSAGE = <<~MESSAGETEXT
+        ■ ダイスロール（ XAS ）
+          ＸＤをロールします。
+          例） 5AS
+        
+        ■ ダイスロール＆最大になるようにピック（ XASY ）
+          ＸＤをロールし、そこから最大になるようにＹ個をピックします。
+          例） 4AS3
+
+        ■ 表
+          ・奇跡の触媒
+            ・エレメント (CELE, CElement)
+            ・アルケミア (CALC, CAlchemia)
+            ・インフォーマント (CINF, CInformant)
+            ・イノセンス (CINN, CInnocence)
+            ・アクワイヤード (CACQ, CAcquired)
+          ・携行品
+            ・Ｓサイズ (ARS, ArticleS)
+            ・Ｍサイズ (ARM, ArticleM)
+            ・Ｌサイズ (ARL, ArticleL)
+          ・ＰＣ情報獲得表 (PCI, PCInformation)
+          ・理由表 (REA, Reason)
+          ・交流表 (ASS, Associate)
+          ・接触のきっかけ表 (CON, Contact)
+      MESSAGETEXT
+
+      ROLL_REG = /^(\d+)AS(\d+)?$/i.freeze
+
+      register_prefix('\\d+AS')
+
+      def eval_game_system_specific_command(command)
+        c = resolve_alias command
+
+        try_roll_alchemia(c) ||
+          roll_tables(c, COMPOSITE_TABLES)
+      end
+
+      def resolve_alias(source_command)
+        ALIAS.each do |key, value|
+          if key.upcase == source_command.upcase || value.upcase == source_command.upcase
+            return value
+          end
+        end
+
+        source_command
+      end
+
+      def try_roll_alchemia(command)
+        if command.match? ROLL_REG
+          match = ROLL_REG.match command
+
+          roll_dice_count = match[1].to_i
+
+          if match[2].nil?
+            # ロールのみ（ピックなし）:
+
+            result = roll_alchemia(roll_dice_count)
+            return make_roll_text(result)
+          else
+            # ロールして最大値をピック:
+
+            pick_dice_count = match[2].to_i
+
+            result = roll_alchemia_and_pick(roll_dice_count, pick_dice_count)
+            return make_roll_and_pick_text(result[:rolled_dices], pick_dice_count, result[:picked_dices])
+          end
+        end
+      end
```

Comment:
```suggestion
      def try_roll_alchemia(command)
        match = ROLL_REG.match(command)
        return nil unless match

        roll_dice_count = match[1].to_i

        if match[2].nil?
          # ロールのみ（ピックなし）:

          result = roll_alchemia(roll_dice_count)
          return make_roll_text(result)
        else
          # ロールして最大値をピック:

          pick_dice_count = match[2].to_i

          result = roll_alchemia_and_pick(roll_dice_count, pick_dice_count)
          return make_roll_and_pick_text(result[:rolled_dices], pick_dice_count, result[:picked_dices])
        end
      end
```

対応外コマンドを弾く場合には、早期リターンが望ましいです